src/ipz_parser.cpp

```diff
@@ -625,4 +625,216 @@ types::DbusVariantType IpzVpdParser::readKeywordFromHardware(
     return types::DbusVariantType{
         getKeywordValueFromRecord(l_record, l_keyword, l_recordOffset)};
 }
+
+void IpzVpdParser::updateRecordECC(const auto& i_recordDataOffset,
+                                   const auto& i_recordDataLength,
+                                   const auto& i_recordECCOffset,
+                                   size_t i_recordECCLength,
+                                   types::BinaryVector& io_vpdVector)
+{
+    auto l_recordDataBegin = std::next(io_vpdVector.begin(),
```

Comment:
nit : instead of Begin  maybe could have used StartOffset , but I am good , just an observation

Reply:
why would it differ ? As per my understanding the start offset of the vector or file remains same as the vector is the file contents

Reply:
I meant m_startOffset.. what ever you said is right.. record's offset and record's ecc offset is same for both file and vector as it is taken from EEPROM and not hardcoded in the code .